exercises/Toluwalase_Agoro/ex01_basics/src/main.cpp

```diff
@@ -35,99 +68,83 @@ int main() {
   and gets input from user so a to open
   input file
   */
-  std::vector <std::string> vectorOfWords{};
-  std::vector <int> vectorOfCounts{};
-  std::vector <int> vectorOfOrderToRead{};
-  char filePathToInput[100];
+  
+
+  char filePathToInput[150];
   std::cout << "Please input a file path to input file :";
-  std::cin.getline(filePathToInput, 100);
+  std::cin.getline(filePathToInput, 150);
   std::ifstream fileToRead;
   fileToRead.open(filePathToInput);
+  std::vector <std::pair<int, std::string>> vectorOfCountsAndWords;
 
   if (!fileToRead) {
     /* This if block checks if file was opened correctly
-    and if it was not program is terminated
+    and if not program is terminated
     */
     std::cout << "File could not be opened" << std::endl;
     std::cout << "Ensure that file path is valid" << std::endl;
   }
+  std::string holderString;
+  auto myPredictor = [&holderString](std::pair<int, std::string>& pairOfCountAndWord)
+  {
+    return (pairOfCountAndWord.second == holderString);
+  };
+
   while (!fileToRead.eof()) {
     /* This while block reads input file and adds formatted words to
     VectorOfWords until the end of file has been reached. This block
     has an if , else if block that determines whether word is more than
     4 letters and already exists in VectorOfWords. if it does it increments
     the count in VectorOfCounts otherwise it is added to VectorOfWords
     */
-    std::string holderString{ };
+    
     holderString = getWord(fileToRead);
-    auto location = std::find(vectorOfWords.begin(), vectorOfWords.end(), holderString);
-    if (holderString.size() > 4 && location == vectorOfWords.end()) {
-      vectorOfWords.push_back(holderString);
-      vectorOfCounts.push_back(1);
+    auto location = std::find_if(vectorOfCountsAndWords.begin(),vectorOfCountsAndWords.end(), myPredictor);
+    if (holderString.size() > 4 && location == vectorOfCountsAndWords.end()) {
```

Comment:
Please can you pull the if size>4 check out into a surrounding if statement so you don't need to repeat it here and L107. I think the check on the location can then be turned into an if...else statement without any conditions on the else part.